### PR TITLE
Contextual record analyzer

### PR DESCRIPTION
**Pull request by Noggog**
- Renamed Skyrim projects to `Mutagen.Bethesda.Analyzers.Skyrim`
- Added parameter DTOs for the analyzer interfaces, to avoid needing to change their footprint in the future.
- Changed existing `MajorRecordAnalyzer` systems to be named `IsolatedRecordAnalyzer`
- Added `ContextualRecordAnalyzer`, which has access to `ILinkCache` and `ILoadOrder`

Should help write the analyzer for Npcs that need to look up their races

### CODE REVIEW

File: Mutagen.Bethesda.Analyzers.Testing/ParamExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿using Mutagen.Bethesda.Analyzers.SDK.Analyzers;
+using Mutagen.Bethesda.Plugins.Records;
+
+namespace Mutagen.Bethesda.Analyzers.Testing
+{
+    public static class ParamExtensions
```
Comment (erri120) [inline Mutagen.Bethesda.Analyzers.Testing/ParamExtensions.cs:6]: Should maybe go into the `TestingUtils` project which we can publish on nuget for others to use

File: Mutagen.Bethesda.Analyzers.Testing/TestModule.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+﻿using System.IO.Abstractions;
+using Autofac;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Mutagen.Bethesda.Analyzers.Autofac;
+
+namespace Mutagen.Bethesda.Analyzers.Testing
+{
+    public class TestModule : Module
```
Comment (erri120) [inline Mutagen.Bethesda.Analyzers.Testing/TestModule.cs:9]: Same with this

File: Mutagen.Bethesda.Analyzers.Testing/ParamExtensions.cs (hunk 1 above)
Comment (Noggog) [reply Mutagen.Bethesda.Analyzers.Testing/ParamExtensions.cs:6]: Do we want to just merge the `Mutagen.Bethesda.Analyzers.TestingUtils` project with `Mutagen.Bethesda.Analyzers.Testing`?  Seems to have the same goal

File: Mutagen.Bethesda.Analyzers.Testing/TestModule.cs (hunk 2 above)
Comment (Noggog) [reply Mutagen.Bethesda.Analyzers.Testing/TestModule.cs:9]: Migrating over `TestModule` would mean `Mutagen.Bethesda.Analyzers.TestingUtils` would need to import `AutoFac`, which shouldn't be enforced on anyone downstream.

Autofac is used in the engine library, but that's not meant to be published and imported by people

**Comment by Noggog** [comment]
I melded the IContextualRecordAnalyzer and IIsolatedRecordAnalyzers.   No need for the distinction for now.  

Can probably detect if an analyzers uses the link cache, if that's interesting information

**Comment by Noggog** [comment]
Migrated up one of the ParamExtensions.  Going to bed, but feel free to merge in if you want to work off the new param systems on anything

**Pull request merged**